src/markov/mc_tools.jl

```diff
@@ -291,189 +291,155 @@ todense(::Type, A::Array) = A
-
-##### Arguments
+### Arguments
 
 - `mc::MarkovChain` : MarkovChain instance.
-- `ts_length::Int` : Length of each simulation.
-- `init::Vector{Int}` : Vector containing initial states.
-- `;num_reps::Int(1)` : Number of repetitions of simulation for each element
-of `init`.
-
-##### Returns
-
-- `X::Matrix{Int}` : Array containing the sample paths as columns, of shape
-(ts_length, k), where k = length(init)* num_reps.
+- `ts_length::Int` : Length of simulation
+- `;init::Int=rand(1:n_states(mc))` : Initial state 
 
-"""
-function simulate(mc::MarkovChain, ts_length::Int, init::Vector{Int};
-                  num_reps::Int=1)
-    k = length(init)*num_reps
-    X = Array(Int, ts_length, k)
-    X[1, :] = repmat(init, num_reps)
-
-    simulate!(mc, X)
-    return X
-end
+### Returns
 
+- `X::Vector` : Vector containing the sample path, with length 
+ts_length
 """
-Simulate time series of state transitions of the Markov chain `mc`.
-
-##### Arguments
-
-- `mc::MarkovChain` : MarkovChain instance.
-- `ts_length::Int` : Length of each simulation.
-- `init::Int` : Initial state.
-- `;num_reps::Int(1)` : Number of repetitions of simulation.
-
-##### Returns
 
-- `X::Matrix{Int}` : Array containing the sample paths as columns, of shape
-(ts_length, k), where k = num_reps.
+function simulate(mc::MarkovChain, ts_length::Int;
+                  init::Int=rand(1:n_states(mc)))
+    X = Array(eltype(mc.state_values), ts_length, 1)
+    simulate!(mc, X; init=init)
+    return vec(X)
+end
 
-"""
-simulate(mc::MarkovChain, ts_length::Int, init::Int; num_reps::Int=1) =
-    simulate(mc, ts_length, [init], num_reps=num_reps)
 
 """
-Simulate time series of state transitions of the Markov chain `mc`.
+Fill `X` with sample paths of the Markov chain `mc` as columns.
+The resulting matrix has the state values of `mc` as elements.
 
-##### Arguments
+### Arguments
 
 - `mc::MarkovChain` : MarkovChain instance.
-- `ts_length::Int` : Length of each simulation.
-- `;num_reps::Union{Int, Void}(nothing)` : Number of repetitions of simulation.
-
-##### Returns
-
-- `X::Matrix{Int}` : Array containing the sample paths as columns, of shape
-(ts_length, k), where k = num_reps.
-
+- `X::Matrix` : Preallocated matrix to be filled with sample paths
+of the Markov chain `mc`. The element types in `X` should be the 
+same as the type of the state values of `mc`
+- `;init=rand(1:n_states(mc))` : Can be one of the following
+    - blank: random initial condition for each chain
+    - scalar: same initial condition for each chain
+    - vector: cycle through the elements, applying each as an
+      initial condition
 """
-function simulate(mc::MarkovChain, ts_length::Int; num_reps::Int=1)
-    init = rand(1:size(mc.p, 1), num_reps)
-    return simulate(mc, ts_length, init)
-end
 
-"""
-Fill `X` with sample paths of the Markov chain `mc` as columns.
-
-##### Arguments
+function simulate!(mc::MarkovChain, X::Matrix;
```

Comment:
It is convention in Julia that the argument that is to be changed in a mutating function is the first argument of that function (e.g. see `rand!`, `copy!`, `fill!`, `A_mul_B!`, `A_mul_Bt!`, `A_mul_Bc!`, `At_mul_B!`, `Ac_mul_B!`, `At_mul_Bt!`, `Ac_mul_Bc!`, etc...)

I think we should do that here, so we'd move `X` to be the first argument. Also It makes sense to me to have the type be `X::Union{AbstractVector,AbstractMatrix}`